frontend/packages/dev-console/src/utils/resource-label-utils.ts

```diff
@@ -44,10 +44,9 @@ export const getAppLabels = ({
 };
 
 export const getGitAnnotations = (gitURL: string, gitRef?: string) => {
-  const ref = gitRef || 'master';
   return {
     'app.openshift.io/vcs-uri': gitURL,
-    'app.openshift.io/vcs-ref': ref,
+    'app.openshift.io/vcs-ref': gitRef || '',
```

Comment:
Why do we not default here to master (or main?) anymore in the annotations? This is not required as part of this change, or?

Reply:
This creates problems for repos where the default branch is not master. For ex: https://github.com/redhat-developer/s2i-dotnetcore-ex or https://github.com/eggheadio/egghead-next
The edit icon then gets this incorrect branch name and adds it to the url, which leads to 404. I think that not having a branch in url is better than assuming an incorrect branch.
But I can undo this if it is not needed.

Reply:
Sounds valid for me. You can keep this.

I thought the gitRef was set in these cases. Can you check if we "lost" the branch info in the add flow for these repos?

Reply:
Looks like we do not try to fetch the the default branch if the ref isn't provided, so it never gets set in the first place:
https://github.com/openshift/console/blob/master/frontend/packages/git-service/src/services/github-service.ts#L83
The git APIs do not require branch to be supplied, and will operate on the default branch.

So unless a ref is provided by user, it will not be set.

Reply:
ParseBitbucketUrl() seems to provide branch as well which is used for default branch if a ref is not provided (https://github.com/openshift/console/blob/a07aa74a966fb75a16ecbe86a73fe0e01a86d497/frontend/packages/git-service/src/services/bitbucket-service.ts#L37)

Reply:
It appears to extract branch name from the URL, and return 'master' if there is no branch.

https://github.com/codefresh-io/parse-bitbucket-url/blob/master/index.js#L189